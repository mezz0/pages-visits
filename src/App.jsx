import React from 'react';
import { Switch, Route, withRouter } from 'react-router-dom';
import styled from 'styled-components/macro';
import * as vars from './styles/exports';

// css reset
import './styles/index.css';

// pages
import Home from './pages/home';
import NoMatch from './pages/noMatch';

// components
import Header from './components/header';

const RouteWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  min-width: 100%;
  min-height: 100vh;
  font-family: 'SunRegular', Helvetica, Arial;
  background-color: ${vars.colors.white};
  line-height: 1.3;
`;

const BodyWrapper = styled.div`
  width: 100%;
`;

function App({ location }) {
  return (
    <RouteWrapper>
      <Header location={location} />
      <BodyWrapper>
        <Switch location={location}>
          <Route exact path={`/`} component={Home} />;
          <Route component={NoMatch} />;
        </Switch>
      </BodyWrapper>
    </RouteWrapper>
  );
}

export default withRouter(App);
