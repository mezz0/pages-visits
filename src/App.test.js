import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter as Router } from 'react-router-dom';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import App from "./App";
import Home from "./pages/home";
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from "@testing-library/react";
import { mount } from 'enzyme';
import { testData } from './test_data';

Enzyme.configure({ adapter: new Adapter() })

describe('render', () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Router><App /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('Match Snap shot', () => {
    const tree = renderer
      .create(<Router><Home /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('input', () => {
  const wrapper = mount(<Router><Home /></Router>); 

  it('input button should be present', () => {
    expect(wrapper.find('#fileUpload').length).toEqual(1); 
  });
});

describe('simulate button press', () => {
  let file;

  beforeEach(() => {
    file = new File([testData], 'server.log', { type: 'file/log' });
  });

  it('cover photo upload', async () => {
    const { getByTestId } = render(<Router><Home /></Router>);

    let uploader = getByTestId('fileUpload');

    await waitFor(() =>
      fireEvent.change(uploader, {
        target: { files: [file] },
      })
    );
    let fileUpload = document.getElementById('fileUpload');
    expect(fileUpload.files[0].name).toBe('server.log');
    expect(fileUpload.files.length).toBe(1);
  });
})