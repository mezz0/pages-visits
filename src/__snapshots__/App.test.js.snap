// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Match Snap shot 1`] = `
<div
  className="home__Wrapper-sc-14vzjrl-0 FgCiJ"
>
  <div
    className="buttonBlock"
  >
    <input
      accept=".log"
      data-testid="fileUpload"
      id="fileUpload"
      onChange={[Function]}
      type="file"
    />
  </div>
</div>
`;
